### PR TITLE
refactoring; improve documentation; more consistent and precise names

### PR DESCRIPTION
**Pull request by mikez**

### CODE REVIEW

**Comment by mikez** [comment]
@AlexanderWillner I rewrote your search code to be more consistent with the rest and added documentation. Clear examples can save a lot of time without having to dig into subcodes.

Might you check if your function does what you intended it to do? Right now it seems to support `None`, `True`, `False` as inputs. Maybe we could remove those options?

**Comment by AlexanderWillner** [comment]
> I rewrote your search code to be more consistent with the rest and added documentation. Clear examples can save a lot of time without having to dig into subcodes.

Thank you, looks very good.

> Might you check if your function does what you intended it to do? 

It does.

> Right now it seems to support None, True, False as inputs. Maybe we could remove those options?

Ah, yeah, this can be removed, only a string is expected.

File: things/api.py [hunk 1]
```diff
@@ -90,9 +90,10 @@ def tasks(uuid=None, include_items=False, **kwargs):  # noqa: C901
         If the argument is `True`, only include tasks _with_ a due date.
         If the argument is `None` (default), then include all tasks.
 
-    search_query : str, bool, or None, optional
+    search_query : str, optional
```
Comment (AlexanderWillner) [inline things/api.py:93]: I think it's not `optional`

File: things/database.py [hunk 2]
```diff
@@ -676,59 +676,16 @@ def make_search_filter(query: str) -> str:
...
-    >>> make_sub_search_filter('TASK.title', True)
-    'TASK.title IS NOT NULL'
-    """
-    default = f'{column} LIKE "%{query}%"'
-    return {
-        None: "",
-        False: f"{column} IS NULL",
-        True: f"{column} IS NOT NULL",
-    }.get(query, default)
+    return f"AND ({' OR '.join(sub_searches)})"
```
Comment (AlexanderWillner) [inline things/database.py:688]: much better

File: things/api.py (hunk 1 above)
Comment (mikez) [reply things/api.py:93]: It is optional. Remember, we are in the `tasks` function here.

**Pull request merged**